src/sql/src/plan/query.rs

```diff
@@ -3120,27 +3124,49 @@ pub fn resolve_func(
     bail!("function {}({}) does not exist", name, types.join(", "))
 }
 
-fn plan_is_null_expr<'a>(
+fn plan_is_expr<'a>(
     ecx: &ExprContext,
     inner: &'a Expr<Aug>,
+    construct: IsExprConstruct,
     not: bool,
 ) -> Result<HirScalarExpr, anyhow::Error> {
-    // PostgreSQL can plan `NULL IS NULL` but not `$1 IS NULL`. This is at odds
-    // with our type coercion rules, which treat `NULL` literals and
-    // unconstrained parameters identically. Providing a type hint of string
-    // means we wind up supporting both.
-    let expr = plan_expr(ecx, inner)?.type_as_any(ecx)?;
-    let mut expr = HirScalarExpr::CallUnary {
-        func: UnaryFunc::IsNull(expr_func::IsNull),
+    let planned_expr = plan_expr(ecx, inner)?;
+    let expr = if construct.requires_boolean_expr() {
+        let expr_coerced= typeconv::plan_coerce(ecx, planned_expr, &ScalarType::Bool)?;
+        let expr_ty = ecx.scalar_type(&expr_coerced);
+        // PostgreSQL allows booleans themselves or string representations of booleans only.
+        // Even though we can coerce an integer to a boolean, this should not succeed.
+        if expr_ty == ScalarType::Bool || expr_ty == ScalarType::String {
+            expr_coerced
+        } else {
+            let not_msg = if not { "NOT " } else { "" };
```

Comment:
minor: I think this should be `" NOT"` instead of `"NOT "` based on the error string below. The way its currently written seems like it would print something like:

```
"ISNOT TRUE requires boolean expression"
```